integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonOutputFormat.java

```diff
@@ -18,43 +18,115 @@
-    org.apache.hadoop.mapreduce.JobContext jobContext = Job.getInstance(jobConf);
-    super.checkOutputSpecs(jobContext);
+  public void checkOutputSpecs(FileSystem fileSystem, JobConf jobConf) throws IOException {
   }
 
   @Override
   public FileSinkOperator.RecordWriter getHiveRecordWriter(JobConf jc, Path finalOutPath,
       Class<? extends Writable> valueClass, boolean isCompressed, Properties tableProperties,
-      Progressable progress) {
-    return null;
+      Progressable progress) throws IOException {
+    CarbonLoadModel carbonLoadModel = null;
+    String encodedString = jc.get(LOAD_MODEL);
+    if (encodedString != null) {
+      carbonLoadModel =
+          (CarbonLoadModel) ObjectSerializationUtil.convertStringToObject(encodedString);
+    }
+    if (carbonLoadModel == null) {
+      carbonLoadModel = HiveCarbonUtil.getCarbonLoadModel(tableProperties, jc);
+    } else {
+      for (Map.Entry<Object, Object> entry : tableProperties.entrySet()) {
+        carbonLoadModel.getCarbonDataLoadSchema().getCarbonTable().getTableInfo().getFactTable()
+            .getTableProperties().put(entry.getKey().toString().toLowerCase(),
+            entry.getValue().toString().toLowerCase());
+      }
+    }
+    String tablePath = FileFactory.getCarbonFile(carbonLoadModel.getTablePath()).getAbsolutePath();
+    TaskAttemptID taskAttemptID = TaskAttemptID.forName(jc.get("mapred.task.id"));
+    TaskAttemptContextImpl context = new TaskAttemptContextImpl(jc, taskAttemptID);
+    final boolean isHivePartitionedTable =
+        carbonLoadModel.getCarbonDataLoadSchema().getCarbonTable().isHivePartitionTable();
+    PartitionInfo partitionInfo =
+        carbonLoadModel.getCarbonDataLoadSchema().getCarbonTable().getPartitionInfo();
+    final int partitionColumn =
+        partitionInfo != null ? partitionInfo.getColumnSchemaList().size() : 0;
+    String finalOutputPath = FileFactory.getCarbonFile(finalOutPath.toString()).getAbsolutePath();
+    if (carbonLoadModel.getCarbonDataLoadSchema().getCarbonTable().isHivePartitionTable()) {
+      carbonLoadModel.getOutputFilesInfoHolder().addToPartitionPath(finalOutputPath);
+      context.getConfiguration().set("carbon.outputformat.writepath", finalOutputPath);
+    }
+    CarbonTableOutputFormat.setLoadModel(context.getConfiguration(), carbonLoadModel);
+    org.apache.hadoop.mapreduce.RecordWriter<NullWritable, ObjectArrayWritable> re =
+        super.getRecordWriter(context);
+    return new FileSinkOperator.RecordWriter() {
+      @Override
+      public void write(Writable writable) throws IOException {
+        try {
+          ObjectArrayWritable objectArrayWritable = new ObjectArrayWritable();
+          if (isHivePartitionedTable) {
+            Object[] actualRow = ((CarbonHiveRow) writable).getData();
+            Object[] newData = Arrays.copyOf(actualRow, actualRow.length + partitionColumn);
+            String[] partitionValues = finalOutputPath.substring(tablePath.length()).split("/");
```

Comment:
what if the partition value itself contains "/" or "=" ? can happen if partition table is string column !

Reply:
/ in partition would be converted to '%2' when creating the partition folder, so it wount fail.

Reply:
yeah ok, I remember now